site_scons/daos_build.py

```diff
@@ -78,7 +79,17 @@ def library(env, *args, **kwargs):
     denv = env.Clone()
     denv.Replace(RPATH=[])
     add_rpaths(denv, kwargs.get('install_off', '..'), False, False)
-    return denv.SharedLibrary(*args, **kwargs)
+    lib = denv.SharedLibrary(*args, **kwargs)
+    if 'target' in kwargs:
+        libname = os.path.basename(kwargs['target'])
+    else:
+        libname = os.path.basename(args[0])
+    if libname.startswith('lib'):
+        libname = libname[3:]
+    varname = f"{libname}_lib"
+    globals()[varname] = lib
+    Export(varname)
+    return lib
```

Comment:
This seems like a bit of a hack, I think the correct way would be to have each place this is called from export as necessary.